lib/passes/MustTailDeadArgElimination.cpp

```diff
@@ -0,0 +1,1137 @@
+//===- DeadArgumentElimination.cpp - Eliminate dead arguments -------------===//
```

Comment:
This is a huge file that came from the LLVM code base. Could you give me some pointers to the code that you added so that I can focus my review on that?

Reply:
My advice is just to check out the file locally and do a diff with the version in llvm 16.0.6. I don't have another way of showing you the exact difference between these two versions. The actual operational difference is that tailcc musttail calls do not inhibit the optimization.

Reply:
OK I did what you suggested and the changes seem reasonable to me.